File: BM28_二叉树的最大深度/javascript_solutions/queue_solution.ts
class TreeNode {
    val: number
    left: TreeNode | null
    right: TreeNode | null
    constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
        this.val = (val === undefined ? 0 : val)
        this.left = (left === undefined ? null : left)
        this.right = (right === undefined ? null : right)
    }
}

/**
 * 代码中的类名、方法名、参数名已经指定，请勿修改，直接返回方法规定的值即可
 *
 *
 * @param root TreeNode类
 * @return int整型
 */
export function maxDepth(root: TreeNode): number {
    if (root == null) return 0;

    let result = 0;

    const queue = [root];

    while (queue.length) {
        const size = queue.length;

        for (let index = 0; index < size; index++) {
            const node = queue.pop();

            if(node?.left) queue.unshift(node?.left);
            if(node?.right) queue.unshift(node?.right);
        }

        result++;
    }

    return result;
}

File: BM28_二叉树的最大深度/python_solutions/queue_solution.py
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

# 代码中的类名、方法名、参数名已经指定，请勿修改，直接返回方法规定的值即可
#
# 
# @param root TreeNode类 
# @return int整型
#
class Solution:
    def maxDepth(self , root: TreeNode) -> int:
        if root == None: return 0
        
        result = 0
        queue = [root]

        while len(queue) > 0:
            size = len(queue)

            for _ in range(size):
                node = queue.pop(0)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)

            result += 1

        return result